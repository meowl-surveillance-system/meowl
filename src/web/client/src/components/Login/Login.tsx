import React, { ComponentState, Component, ChangeEvent } from 'react';
import { Redirect } from 'react-router-dom';
import { Container, FormControl, TextField, Button, Typography } from '@material-ui/core';

interface Props {
  isLoggedIn: boolean
}

interface State {
  username: string;
  password: string;
}

export default class Login extends Component<Props, State> {

  constructor(props: Props) {
    super(props);
    this.state = {
      username: "",
      password: "",
    };
  }

  handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
    let target: HTMLInputElement = (event.target as HTMLInputElement);
    let label: string = target.name;
    let value: string = target.value;
    this.setState({ [label]: value } as ComponentState);
  };

  loginSubmit = (): void => {
    const requestOptions = {
      method: "POST",
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({username:this.state.username, password:this.state.password})
    };
    fetch(`/auth/login`, requestOptions).then(res => {
      this.setState({username:"" + res.status});
    });
  }
  logoutSubmit = (): void => {
    const requestOptions = {
      method: "POST",
    };
    fetch(`/auth/logout`, requestOptions).then(res => {
      this.setState({username:"" + res.status});
    });
  }

  render() {
    if(this.props.isLoggedIn) {
      return <Redirect to="/" />
    }
    return (
      <Container>
        <Typography variant="h5" component="h6">Login</Typography>
        <FormControl data-testid="form">
          <TextField
            id="username-input"
            name="username"
            value={this.state.username}
            onChange={this.handleChange}
            placeholder="Username"
            inputProps={{ "data-testid": "username-test" }}
          />
          <TextField
            id="password-input"
            name="password"
            value={this.state.password}
            onChange={this.handleChange}
            placeholder="Password"
	    type="password"
            inputProps={{ "data-testid": "password-test" }}
          />
          <Button
            onClick={() => this.loginSubmit()}
          >
            Login
          </Button>
          <Button
            onClick={() => this.logoutSubmit()}
          >
            Logout
          </Button>
        </FormControl>
      </Container >
    );
  }
}
