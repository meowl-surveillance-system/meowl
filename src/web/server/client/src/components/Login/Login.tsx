import React, { ComponentState, Component, ChangeEvent } from 'react';
import ReactPlayer from 'react-player';
import { Container, FormControl, TextField, Button, Typography } from '@material-ui/core';

interface Props { }

interface State {
  username: string;
  password: string;
}

export default class Login extends Component<Props, State> {

  constructor(props: Props) {
    super(props);
    this.state = {
      username: "",
      password: "",
    };
  }

  handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
    let target: HTMLInputElement = (event.target as HTMLInputElement);
    let label: string = target.name;
    let value: string = target.value;
    this.setState({ [label]: value } as ComponentState);
  };

  loginSubmit = (): void => {
    const requestOptions = {
      method: "POST",
      body: JSON.stringify({"username":this.state.username, "password":this.state.password})
    };
    fetch(`/auth/login`, requestOptions).then(res => {
      this.setState({username:"" + res.status});
    });
  }

  render() {
    return (
      <Container>
        <Typography variant="h5" component="h6">Login</Typography>
        <FormControl data-testid="form">
          <TextField
            id="username-input"
            name="username"
            value={this.state.username}
            onChange={this.handleChange}
            placeholder="Username"
            inputProps={{ "data-testid": "username-test" }}
          />
          <TextField
            id="password-input"
            name="password"
            value={this.state.password}
            onChange={this.handleChange}
            placeholder="Password"
            inputProps={{ "data-testid": "password-test" }}
          />
          <Button
            onClick={() => this.loginSubmit()}
          >
            Login
          </Button>
        </FormControl>
      </Container >
    );
  }
}
