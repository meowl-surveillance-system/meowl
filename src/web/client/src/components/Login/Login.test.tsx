import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Login from "./Login";

describe("Login component", () => {
  it("renders Login component", () => {
    const historyMock: object = {
      push: jest.fn(),
    };
    const onAuthChangeMock: jest.Mock = jest.fn();
    const { container } = render(
      <Login
        isLoggedIn={false}
        history={historyMock}
        onAuthChange={onAuthChangeMock}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
