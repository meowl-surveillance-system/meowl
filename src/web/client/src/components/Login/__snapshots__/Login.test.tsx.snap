// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders Login component 1`] = `
<main
  class="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    style="min-height: 100vh;"
  >
    <h6
      class="MuiTypography-root MuiTypography-h5"
    >
      Login
    </h6>
    <form
      novalidate=""
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="username-test"
            id="username-input"
            name="username"
            placeholder="Username"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-180 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-181"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="password-test"
            id="password-input"
            name="password"
            placeholder="Password"
            required=""
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-180 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-181"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Login
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</main>
`;
